x/concentrated-liquidity/msg_server_test.go

```diff
@@ -388,36 +389,75 @@ func (s *KeeperTestSuite) TestCollectIncentives_Events() {
-			position, err := s.App.ConcentratedLiquidityKeeper.GetPosition(ctx, tc.positionIds[0])
+			position, err := s.App.ConcentratedLiquidityKeeper.GetPosition(s.Ctx, tc.positionIds[0])
 			s.Require().NoError(err)
-			ctx = ctx.WithBlockTime(position.JoinTime.Add(time.Hour * 24 * 7))
-			positionAge := ctx.BlockTime().Sub(position.JoinTime)
+			s.Ctx = s.Ctx.WithBlockTime(position.JoinTime.Add(time.Hour * 24 * 7))
+			positionAge := s.Ctx.BlockTime().Sub(position.JoinTime)
 
 			// Set up accrued incentives
-			err = addToUptimeAccums(ctx, pool.GetId(), s.App.ConcentratedLiquidityKeeper, uptimeHelper.hundredTokensMultiDenom)
+			err = addToUptimeAccums(s.Ctx, pool.GetId(), s.App.ConcentratedLiquidityKeeper, uptimeHelper.hundredTokensMultiDenom)
 			s.Require().NoError(err)
-			s.FundAcc(pool.GetIncentivesAddress(), expectedIncentivesFromUptimeGrowth(uptimeHelper.hundredTokensMultiDenom, DefaultLiquidityAmt, positionAge, sdk.NewInt(int64(len(tc.positionIds)))))
+
+			expectedIncentives := sdk.Coins{}
+			expectedForfeit := sdk.Coins{}
+
+			// Determine uptime growth at time of claim
+			// This isn't really straight forward, since when each position claimed, it forfeits its respective rewards and they become claimable by the next
+			// This modifies the expected uptime accumulator for each position, so we need to determine what that uptime accumulator value will be at the time each position claims.
+			perPosUptimeGrowthAtTimeOfClaim := [][]sdk.DecCoins{}
+			for i := range tc.positionIds {
+				if i == 0 {
+					perPosUptimeGrowthAtTimeOfClaim = append(perPosUptimeGrowthAtTimeOfClaim, uptimeHelper.hundredTokensMultiDenom)
+				} else {
+					newUptimeGrowthPostPreviousUserForfeit := make([]sdk.DecCoins, len(uptimeHelper.hundredTokensMultiDenom))
+					copy(newUptimeGrowthPostPreviousUserForfeit, uptimeHelper.hundredTokensMultiDenom)
+					newUptimeGrowthWithPreviousUserForfeit := perPosUptimeGrowthAtTimeOfClaim[i-1][len(perPosUptimeGrowthAtTimeOfClaim[i-1])-1].MulDec(sdk.NewDec(int64(len(tc.positionIds))).Quo(sdk.NewDec(int64(len(tc.positionIds) - 1))))
+					newUptimeGrowthPostPreviousUserForfeit[len(newUptimeGrowthPostPreviousUserForfeit)-1] = newUptimeGrowthWithPreviousUserForfeit
+					perPosUptimeGrowthAtTimeOfClaim = append(perPosUptimeGrowthAtTimeOfClaim, newUptimeGrowthPostPreviousUserForfeit)
+				}
+			}
+
+			// Using the above calculated uptime growth for each position, calculate the expected redeemed incentives and forfeits.
+			for _, uptimeGrowth := range perPosUptimeGrowthAtTimeOfClaim {
+				expectedIncentivesInternal := expectedIncentivesFromUptimeGrowth(uptimeGrowth, DefaultLiquidityAmt, positionAge, sdk.NewInt(1))
+				expectedForfeitInternal := expectedIncentivesFromUptimeGrowth(uptimeGrowth, DefaultLiquidityAmt, twoWeeks, sdk.NewInt(1)).Sub(expectedIncentivesInternal)
+				expectedIncentives = expectedIncentives.Add(expectedIncentivesInternal...)
+				expectedForfeit = expectedForfeit.Add(expectedForfeitInternal...)
+
+			}
```

Comment:
Is it possible to create a separate test for `ClaimAllIncentivesForPosition` for testing this edge case instead of packaging it into the message server level of abstraction test?

Reply:
Yeah, the msg server level of abstraction test was done since this was done to correct the initial mutation test error, but agreed I can get more tests in for the lower lever func